src/openms/include/OpenMS/FILTERING/ID/IDFilter.h

```diff
@@ -336,6 +336,62 @@ namespace OpenMS
     /// Is the list of peptide evidences of this peptide hit empty?
     struct HasNoEvidence;
 
+    
+    /**
+       @brief Filter Peptide Hit by its digestion product
+
+    */
+
+    struct PeptideDigestionFilter
+    {
+
+      typedef PeptideHit argument_type;
+
+      EnzymaticDigestion& digestion_;
+      Int min_cleavages_;
+      Int max_cleavages_;
+      
+      PeptideDigestionFilter(
+        EnzymaticDigestion& digestion) :  
+      digestion_(digestion)
+      {}
+
+      inline void setMinCleavagesAllowed(const Int min_cleavages){ min_cleavages_ = min_cleavages; }
+      inline void setMaxCleavagesAllowed(const Int max_cleavages){ max_cleavages_ = max_cleavages; }
+
+      inline void disableCleavagesLowerBound(){ min_cleavages_ = disabledValue(); }
+      inline void disableCleavagesUpperBound(){ max_cleavages_ = disabledValue(); }
+
+      inline Int disabledValue(){ return -1; }
+      
+      /// Filter function on min max cutoff values 
+      bool operator()(PeptideHit& p)
+      {
+        return digestion_.filterByMissingCleavages(
+          p.getSequence().toString(), 
```

Comment:
I believe this must be `p.getSequence().toUnmodifiedString()`!! Otherwise there will be issues with modifications.
Please add a test-case.

Reply:
Good catch! I added some tests for that case though I can not quite see why this would not work.